src/components/molecules/FormFields/CustomProvider.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ReactElement, useState, useEffect } from 'react'
+import { useField } from 'formik'
+import { toast } from 'react-toastify'
+import CustomInput from './URLInput/Input'
+import { useOcean } from '../../../providers/Ocean'
+import { InputProps } from '../../atoms/Input'
+
+export default function CustomProvider(props: InputProps): ReactElement {
+  const [field, meta, helpers] = useField(props.name)
+  const [isLoading, setIsLoading] = useState(false)
+  const [providerUrl, setProviderUrl] = useState<string>()
+  const { ocean, config } = useOcean()
```

Comment:
this gets more tricky as I found after my Friday publishing investigation. Basically `ocean` is only initiated automatically in the asset details view so this `ocean` here will be `null`, just as `config`. While the `OceanProvider` now also wraps the publish form I did not add auto-connection for it here and initiate it only within the `usePublish` hook, so only after user hits _Submit_: https://github.com/oceanprotocol/market/blob/main/src/hooks/usePublish.ts#L47-L58

Now, if we require this in more components on publish form as this looks like here, it would make sense to initiate `ocean` in some higher levels like in `src/components/pages/Publish/index.tsx`.

(Note that this all compares to latest `main` so be sure to merge/rebase before continuing here)

Reply:
I'm not sure about this, I did a console.log of ocean here: https://github.com/oceanprotocol/market/blob/cbdb60edb8b87b38dc635182e0f782a0ecd0690d/src/components/molecules/FormFields/CustomProvider.tsx#L13

And it is defined: 
![ocean defined custom provider](https://user-images.githubusercontent.com/20739535/128327546-623a267a-7a80-4356-af1e-7f157de118c8.png)

Reply:
It seems to work as it is, let me know if I'm missing anything?

Reply:
can confirm, all good. This all works on publish page since #768 was merged